test/system/280-update.bats

```diff
@@ -161,4 +161,552 @@ device-write-iops   = /dev/zero:4000 | - | -
     run_podman rm -f -t0 testctr
 }
 
+function _create_container_check_change_of_HealthCheck_configuration_with_podman_update(){
+    local ctrname="$1"
+    local msg="$2"
+    local format="$3"
+    local flag="$4"
+    local value="$5"
+    local expect="$6"
+    local expect_msg="$7"
+    local expect_exit_code="$8"
+    local is_startup="$9"
+    local no_hc="${10}"
```

Comment:
These can be collapsed into a `type` option (choose a better name please) with values `""` (third condition), `"nohc"` (second), `"startup"` (first).